src/containers/org/components/DeleteOrgModal.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * @flow
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { ActionModal } from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState } from 'lattice-utils';
+import { useDispatch } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type { UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import { ModalBody } from '../../../components';
+import { resetRequestState } from '../../../core/redux/actions';
+import { ORGANIZATIONS } from '../../../core/redux/constants';
+import { DELETE_EXISTING_ORGANIZATION, deleteExistingOrganization } from '../actions';
+
+const { isSuccess } = ReduxUtils;
+
+type Props = {
+  isOwner :boolean;
+  isVisible :boolean;
+  onClose :() => void;
+  organizationId :UUID;
+};
+
+const DeleteOrgModal = ({
+  isOwner,
+  isVisible,
+  onClose,
+  organizationId,
+} :Props) => {
+
+  const [standbyMessage, updateStandbyMessage] = useState(
+    'Are you sure you want to delete this organization? This action cannot be undone.'
+  );
+
+  const dispatch = useDispatch();
+
+  const deleteOrgRS :?RequestState = useRequestState([ORGANIZATIONS, DELETE_EXISTING_ORGANIZATION]);
+
+  const handleOnClickPrimary = () => {
+    if (isOwner) {
+      dispatch(deleteExistingOrganization(organizationId));
+    }
+    else {
+      updateStandbyMessage('You must be an owner to delete an organization.');
+    }
+  };
+
+  const handleOnClose = () => {
+    onClose();
+    setTimeout(() => {
+      dispatch(resetRequestState([DELETE_EXISTING_ORGANIZATION]));
+    }, 1000);
+  };
```

Comment:
Wrap ResetOnUnmount around success and failure components instead of relying on setTimeout. This is flaky as the user could easily re-open the model within the timeout period.